src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityStateMessage.java

```diff
@@ -0,0 +1,103 @@
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import org.jetbrains.annotations.NotNull;
+
+public class ChangeDeliveryPriorityStateMessage extends AbstractMessage<ChangeDeliveryPriorityStateMessage, IMessage>
+{
+    /**
+     * The Colony ID.
+     */
+    private int colonyId;
+
+    /**
+     * Id of the building.
+     */
+    private BlockPos buildingId;
+
+    /**
+     * The dimension of the message.
+     */
+    private int dimension;
+
+    /**
+     * Empty public constructor.
+     */
+    public ChangeDeliveryPriorityStateMessage()
+    {
+        super();
+    }
+
+    /**
+     *  Creates message for player to change the priority of the delivery.
+     *
+     * @param building        view of the building to read data from
+     */
+    public ChangeDeliveryPriorityStateMessage(@NotNull final AbstractBuildingView building)
+    {
+        super();
+        this.colonyId = building.getColony().getID();
+        this.buildingId = building.getLocation();
+        this.dimension = building.getColony().getDimension();
+    }
+
+    /**
+     * Transformation from a byteStream to the variables.
+     *
+     * @param buf the used byteBuffer.
+     */
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        this.colonyId = buf.readInt();
+        this.buildingId = BlockPosUtil.readFromByteBuf(buf);
+        dimension = buf.readInt();
+    }
+
+    /**
+     * Transformation to a byteStream.
+     *
+     * @param buf the used byteBuffer.
+     */
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        buf.writeInt(this.colonyId);
+        BlockPosUtil.writeToByteBuf(buf, this.buildingId);
+        buf.writeInt(dimension);
+    }
+
+    @Override
+    public void messageOnServerThread(final ChangeDeliveryPriorityStateMessage message, final EntityPlayerMP player)
+    {
+        final Colony colony = ColonyManager.getColonyByDimension(message.colonyId, message.dimension);
+        if (colony != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
+        {
+            //Verify player has permission to change this huts settings
+            if (!colony.getPermissions().hasPermission(player, Action.MANAGE_HUTS))
+            {
+                return;
+            }
+            final AbstractBuilding building = colony.getBuildingManager().getBuilding(message.buildingId);
+
+            if(building instanceof AbstractBuildingWorker)
+            {
+                MineColonies.getLogger().info(building.getPriorityState());

```

Comment:
I think we can remove the logger here now =)

Reply:
woops